examples/mic_vad_streaming/mic_vad_streaming.py

```diff
@@ -0,0 +1,188 @@
+import time, logging
+from datetime import datetime
+import threading, collections, queue, os, os.path
+import wave
+import pyaudio
+import webrtcvad
+from halo import Halo
+import deepspeech
+import numpy as np
+
+logging.basicConfig(level=20)
+
+class Audio(object):
+    FORMAT = pyaudio.paInt16
+    RATE = 16000
+    CHANNELS = 1
+    BLOCKS_PER_SECOND = 50
+    BLOCK_SIZE = int(RATE / float(BLOCKS_PER_SECOND))
+
+    def __init__(self, callback=None):
+        def proxy_callback(in_data, frame_count, time_info, status):
+            callback(in_data)
+            return (None, pyaudio.paContinue)
+        if callback is None: proxy_callback = None
+        self.sample_rate = self.RATE
+        self.block_size = self.BLOCK_SIZE
+        self.pa = pyaudio.PyAudio()
+        self.stream = self.pa.open(format=self.FORMAT,
+                                   channels=self.CHANNELS,
+                                   rate=self.sample_rate,
+                                   input=True,
+                                   frames_per_buffer=self.block_size,
+                                   stream_callback=proxy_callback)
+        self.stream.start_stream()
+
+    def destroy(self):
+        self.stream.stop_stream()
+        self.stream.close()
+        self.pa.terminate()
+
+    frame_duration_ms = property(lambda self: 1000 * self.block_size // self.sample_rate)
+
+    def write_wav(self, filename, data):
+        logging.info("write wav %s", filename)
+        wf = wave.open(filename, 'wb')
+        wf.setnchannels(self.CHANNELS)
+        # wf.setsampwidth(self.pa.get_sample_size(FORMAT))
+        assert FORMAT == pyaudio.paInt16
+        wf.setsampwidth(2)
+        wf.setframerate(self.sample_rate)
+        wf.writeframes(data)
+        wf.close()
+
+    @classmethod
+    def main(cls, callback):
+        audio = cls(proxy_callback)
+        while audio.stream.is_active():
+            time.sleep(0.1)
+        audio.destroy()
+
+class VADAudio(Audio):
+    def __init__(self, consumer, aggressiveness):
+        super().__init__()
+        self.vad = webrtcvad.Vad(aggressiveness)
+        t = threading.Thread(target=consumer, args=(self, self.vad_collector(300),))
```

Comment:
Gah, yes. I was intending to use `pyaudio`'s callback mode, which uses a separate thread, but I accidentally left in the blocking mode from testing. I'll fix. (The flagged thread is actually superfluous since this script does nothing other than the streaming recognition.)

Reply:
Feel free to flag me again for review when it's fixed :). Basically you don't want to perform Streaming API calls on the same thread as audio recording

Reply:
Fixed. `pyaudio` callback runs in a separate thread.